src/go/agwd/config/config.go

```diff
@@ -46,22 +46,25 @@ func LogLevel(l config.AgwD_LogLevel) log.Level {
 
 const (
 	ipv4Scheme = "ipv4"
+	tcp4Scheme = "tcp4"
 	ipv6Scheme = "ipv6"
+	tcp6Scheme = "tcp6"
 )
 
 // ParseTarget takes a target in string form and returns a resolved Target.
 // Extends functionality in grpc/internal/grpcutil.ParseTarget to support ipv4
-// and ipv6 schemes.
+// and ipv6 schemes. Return tcp4Scheme or tcp6Scheme so the Scheme can be passed
+// directly to net.Listen.
 func ParseTarget(target string) resolver.Target {
 	if strings.HasPrefix(target, ipv4Scheme+":") {
 		return resolver.Target{
-			Scheme:   ipv4Scheme,
+			Scheme:   tcp4Scheme,
```

Comment:
i think you need to add handlers for tcp4 and tcp6 schemes.

Reply:
Annoyingly, gRPC and Go's Net library proscribe different formats:
https://github.com/grpc/grpc/blob/master/doc/naming.md
https://pkg.go.dev/net#Dial

gRPC wants "ipv4"
Go wants  "ip4" or "tcp4"

So I stuck with the gRPC convention here and translate when parsing so downstream Go code doesn't have to worry about it.

Reply:
I would recommend returning what gRPC wants and then having a separate helper `func NewTargetListner(target resolver.Target) (net.Listner, error)` that maps scheme to tcp{4,6}